### PR TITLE
configuration & db files for future PRs

### PR DESCRIPTION
**Pull request by zfred219**
- Configuration files to ease future PRs
- See comment below

### CODE REVIEW

File: config/initializers/assets.rb [hunk 1]
```diff
@@ -9,3 +9,8 @@
 # Precompile additional assets.
 # application.js, application.css, and all non-JS/CSS in app/assets folder are already added.
 # Rails.application.config.assets.precompile += %w( search.js )
+# Rails.application.config.assets.precompile += %w( main_page.js )
+Rails.application.config.assets.precompile += %w( apps.scss )
+Rails.application.config.assets.precompile += %w( matching.css )
+Rails.application.config.assets.precompile += %w( progress.css )
+Rails.application.config.assets.precompile += %w( navbar.css )
```
Comment (zfred219) [inline config/initializers/assets.rb:16]: Preprocess scss file to load scss file for app

File: config/initializers/omniauth.rb [hunk 2]
```diff
@@ -10,7 +10,7 @@
                                      :name => user['name'],
                                      :nickname => user['github_uid'],
                                      :email => user['email'],
-                                     :type_user => user['staff']
+                                     :type_user => user['user_type']
```
Comment (zfred219) [inline config/initializers/omniauth.rb:13]: Fixed item name

File: config/routes.rb [hunk 3]
```diff
@@ -1,4 +1,21 @@
...
+  post '/matching/:matching_id/engagement/:engagement_id/store' => 'matching#store'
+  get '/matching/:matching_id/result' => 'matching#result', :as => 'matching_result'
+  delete '/matching/:matching_id' => 'matching#destroy', :as => 'delete_matching'
+  post '/matching/:matching_id/result/final_edit' => 'matching#final_edit'
+  get '/matching/:matching_id/result/finalize' => 'matching#finalize', :as => 'matching_finalize'
+  post '/matching/:matching_id/engagement/create' => 'matching#create_engagement'
+  post '/matching/:matching_id/engagement/:engagement_id/update' => 'matching#update_engagement'
+  delete '/matching/:matching_id/engagement/:engagement_id' => 'matching#delete_engagement', :as => 'delete_engagement'
+  post '/matching/:matching_id/apps/update' => 'matching#update_apps'
+
```
Comment (zfred219) [inline config/routes.rb:18]: All the routes needed for matching-related features

File: config/routes.rb [hunk 4]
```diff
@@ -14,12 +31,20 @@
   resources :engagements, :only => [] do # don't route engagements by themselves
     resources :iterations
   end
+
+  get 'orgs/import' => 'orgs#import'
```
Comment (zfred219) [inline config/routes.rb:35]: For importing csv purpose

File: db/github_mock_login.yml [hunk 5]
```diff
@@ -3,8 +3,22 @@ development:
...
+  github_uid: "esaas_student_github"
+  name: "ESaas Student"
+  email: "esaas_student@saasbook.info"
+  user_type: 'student'
+
+test_client:
+  github_uid: "esaas_client_github"
+  name: "ESaas Client"
+  email: "esaas_client@saasbook.info"
+  user_type: 'client'
```
Comment (zfred219) [inline db/github_mock_login.yml:24]: Add different type of mock user for testing purpose

File: db/migrate/20210303105028_add_semester_to_iterations.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+class AddSemesterToIterations < ActiveRecord::Migration
+  def change
+    add_column :engagements, :semester, :string
+  end
+end
```
Comment (zfred219) [inline db/migrate/20210303105028_add_semester_to_iterations.rb:5]: Add semester field for searching purpose

File: db/migrate/20210328005618_create_matchings.rb [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+      t.text :projects
+      t.text :preferences
+      t.text :description
+      t.text :result
+      t.string :status, null: false, default: 'Collecting responses'
+      t.text :last_edit_users
+      t.timestamps :last_edit_time
+    end
+  end
+end
```
Comment (zfred219) [inline db/migrate/20210328005618_create_matchings.rb:15]: Create Matching table

File: db/seeds.rb [hunk 8]
```diff
@@ -49,6 +55,10 @@
 			e.presentation_url = row[:link_to_presentation_slides]
 			e.prototype_deployment_url = row[:deployment]
 			e.student_names = row[:students] || 'Unknown'
+
+			# say only spring and summer based on csv
+			semester = "#{row[:semester]}"
+			e.semester = (semester[0] == "F" ? "FALL" : "SPRING") + " 20" +semester[1,3]
```
Comment (zfred219) [inline db/seeds.rb:61]: Change semester style to full name

File: config/environments/production.rb [hunk 9]
```diff
@@ -91,11 +91,11 @@
...
-      access_key_id: ENV.fetch('AWS_ACCESS_KEY_ID'),
-      secret_access_key: ENV.fetch('AWS_SECRET_ACCESS_KEY'),
-      s3_region: ENV.fetch('AWS_REGION'),
+      bucket: ENV["S3_BUCKET_NAME"], # ENV.fetch('S3_BUCKET_NAME')
+      access_key_id: ENV["AWS_ACCESS_KEY_ID"], # ENV.fetch('AWS_ACCESS_KEY_ID')
+      secret_access_key: ENV["AWS_SECRET_ACCESS_KEY"], # ENV.fetch('AWS_SECRET_ACCESS_KEY')
+      s3_region: ENV["AWS_REGION"], # ENV.fetch('AWS_REGION')
     },
-    s3_host_name: ENV.fetch('S3_HOST_NAME')
+    s3_host_name: ENV["S3_HOST_NAME"] #ENV.fetch('S3_HOST_NAME')
```
Comment (zfred219) [inline config/environments/production.rb:99]: Update the syntax

**Comment by zfred219** [comment]
Closed as we will include them as needed each time

File: Gemfile [hunk 10]
```diff
@@ -18,7 +19,6 @@ gem 'jbuilder', '~> 2.0'
 gem 'bootstrap_form'
 gem 'paperclip', '~> 5.2.1'
 gem 'aws-sdk', '~> 2.3.0'
-
 group :development do
   # Access an IRB console on exception pages or by using <%= console %> in views
   gem 'web-console', '~> 2.0'
```
Comment (EnjiaXu) [inline Gemfile:24]: Ignore this file (already in a previous PR)

File: Gemfile.lock [hunk 11]
```diff
@@ -372,6 +374,7 @@ DEPENDENCIES
   jquery-rails
   json
   launchy
+  mimemagic (~> 0.3.10)
   omniauth-github
   paperclip (~> 5.2.1)
   pg (< 1.0.0)
```
Comment (EnjiaXu) [inline Gemfile.lock:380]: Ignore this file (already in a previous PR)

File: features/public_search.feature [hunk 12]
```diff
@@ -24,13 +24,13 @@ Background: Logged Out
...
-Scenario: search for an app by name keyword
-    Given I publicly search for "1"
-    Then I should see "app 1"
-    But I should not see "app 2"
-    And I should not see "app 3"
+#Scenario: search for an app by name keyword
+   # Given I publicly search for "1"
+    #Then I should see "app 1"
+   # But I should not see "app 2"
+    #And I should not see "app 3"
```
Comment (EnjiaXu) [inline features/public_search.feature:36]: Ignore this file (already in a previous PR)

File: db/schema.rb [hunk 13]
```diff
@@ -93,6 +95,18 @@
...
+    t.text     "preferences"
+    t.text     "description"
+    t.text     "result"
+    t.string   "status",          default: "Collecting responses", null: false
+    t.text     "last_edit_users"
+  end
+
   create_table "orgs", force: :cascade do |t|
     t.string   "name"
     t.string   "description"
```
Comment (EnjiaXu) [inline db/schema.rb:112]: Schema modified for two features: 
- add semester info to engagements
- new matching feature

**Pull request merged**